File: app/src/main/java/com/iamwent/diary/modules/day/DayListAdapter.java
package com.iamwent.diary.modules.day;

import android.content.Context;
import androidx.annotation.LayoutRes;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.iamwent.diary.R;
import com.iamwent.diary.data.bean.Diary;

import java.util.List;

/**
 * Created by iamwent on 9/19/16.
 *
 * @author iamwent
 * @since 9/19/16
 */
class DayListAdapter extends RecyclerView.Adapter<DayListAdapter.ViewHolder> {

    private Context ctx;
    private List<Diary> diaries;

    DayListAdapter(Context ctx, List<Diary> diaries) {
        this.ctx = ctx;
        this.diaries = diaries;
    }

    private
    @LayoutRes
    int provideItemLayout() {
        return R.layout.item_day;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(ctx).inflate(provideItemLayout(), parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        holder.bind(diaries.get(position));
    }

    @Override
    public int getItemCount() {
        return diaries == null ? 0 : diaries.size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {

        TextView tvDay;

        ViewHolder(View itemView) {
            super(itemView);

            tvDay = (TextView) itemView.findViewById(R.id.tv_day);
        }

        void bind(Diary diary) {
            tvDay.setText(diary.title);
        }
    }
}

File: app/src/main/java/com/iamwent/diary/modules/day/DayListAdapter.kt
package com.iamwent.diary.modules.day

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.annotation.LayoutRes
import androidx.recyclerview.widget.RecyclerView
import com.iamwent.diary.R
import com.iamwent.diary.data.bean.Diary

/**
 * Created by iamwent on 9/19/16.
 *
 * @author iamwent
 * @since 9/19/16
 */
internal class DayListAdapter(private val ctx: Context, private val diaries: List<Diary>?) : RecyclerView.Adapter<DayListAdapter.ViewHolder>() {
    @LayoutRes
    private fun provideItemLayout(): Int {
        return R.layout.item_day
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(ctx).inflate(provideItemLayout(), parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(diaries!![position])
    }

    override fun getItemCount(): Int {
        return diaries?.size ?: 0
    }

    internal class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var tvDay: TextView

        init {
            tvDay = itemView.findViewById<View>(R.id.tv_day) as TextView
        }

        fun bind(diary: Diary) {
            tvDay.text = diary.title
        }
    }
}
